hardware/raspberrypi/spi/README.md

```diff
@@ -3,51 +3,53 @@
 ## Page Contents
 
 - [Overview](#overview)  
-- [Software](#software)
 - [Hardware](#hardware)
+- [Software](#software)
 - [Linux driver](#driver)
 - [Troubleshooting](#troubleshooting)
 
 <a name="overview"></a>
 ## Overview
 
-The Raspberry Pi is equipped with one [SPI](https://en.wikipedia.org/wiki/Serial_Peripheral_Interface_Bus) bus that has 2 chip selects.
-
-The SPI master driver is disabled by default on Raspberry Pi OS. To enable it, use [raspi-config](../../../configuration/raspi-config.md), or ensure the line `dtparam=spi=on` isn't commented out in `/boot/config.txt`, and reboot. If the SPI driver was loaded, you should see the device `/dev/spidev0.0`.
-
-The SPI bus is available on the P1 Header:
-
-```
-MOSI    P1-19
-MISO    P1-21
-SCLK    P1-23   P1-24    CE0
-GND     P1-25   P1-26    CE1
-```
-
-<a name="software"></a>
-## Software
-
-### bcm2835 library
-
-This is a C library for Raspberry Pi (RPi). It provides access to GPIO and other IO functions on the Broadcom BCM 2835 chip. Accesses the hardware registers directly.
-
-http://www.airspayce.com/mikem/bcm2835/
-### Use spidev from C
-
-There's a loopback test program in the Linux documentation that can be used as a starting point. See the [Troubleshooting](#troubleshooting) section. Uses the Linux `spidev` driver to access the bus.
-
-### Shell
-
-```bash
-# Write binary 1, 2 and 3
-echo -ne "\x01\x02\x03" > /dev/spidev0.0
-```
+The Raspberry Pi family of devices is equipped with three [SPI](https://en.wikipedia.org/wiki/Serial_Peripheral_Interface_Bus) buses. The main SPI interface is referred to as SPI0 in the documentation; the second is SPI1. SPI2 is only usable on a Compute Module or Compute Module 3. By default each SPI bus has 2-3 chip selects, but this number can be changed.
```

Comment:
This is definitely the kind of thing that is needed.

One query.  In my original researches on this, I discovered the information that the additional buses were only available on the Compute Module as you've stated.  However, when I said this on the Raspberry Pi Forum a user claimed that only the Pi 4 has more than one Bus.  So which Busses are available on which Pis; as above or something else?

Apropos my last comment, I've seen people talking about adding Overlays to increase the number of buses up to a maximum of six.  (I think because six overlays appear in the overlays directory.)  Again who is right?

Thanks for doing this.

Reply:
SPI0-2 are available on BCM2835, BCM2836, BCM2837. SPI0 and SPI1 are usable on all Pis with 40-pin headers, while SPI2 is only usable on Compute Modules (including CM3).

BCM2711 (as used on Pi 4) adds four new SPI buses - SPI3-SPI6. They deserve a mention here.

All the SPI buses are disabled by default, so all need a dtparam or a dtoverlay to activate them.

Reply:
So can SPI3-6 only be  used on a hypothetical CM4? Or is there more muxing so they can be used on the 40pin header?

Reply:
All pins needed for SPI3-6 appear on the 40-pin header. I'm in the middle of adding that now.

Reply:
Added SPI3-6.